File: Encryption/railFenceDec.cpp
// 전치 암호 2

#include <bits/stdc++.h>
int n;
char rail[10][100];
char d[100], e[100];

int main() {
    scanf("%d ", &n);
    fgets(e, 100, stdin);
    int elen = strlen(e)-1;
    
    bool di;
    int row = 0, col = 0;
    for(int i=0; i<elen; i++) {
        if(row == 0) di = true;
        if(row == n-1) di = false;
        rail[row][col++] = '0';
        di?row++:row--;
    }
    
    int cnt = 0;
    for(int i=0; i<n; i++)
        for(int j=0; j<elen; j++)
            if(rail[i][j] == '0' && cnt < elen)
                rail[i][j] = e[cnt++];
    row = col = 0;
    
    for(int i=0; i<elen; i++) {
        if(row == 0) di = true;
        if(row == n-1) di = false;
        if(rail[row][col] != '0')
            d[i] = rail[row][col++];
        di?row++:row--;
    }
    
    printf("%s\n", d);
}

File: Encryption/railFenceDec.py
n = int(input())
e = input()

rail = [['\0' for _ in range(len(e))] for _ in range(n)]
di = False
row, col = 0, 0

for i in range(len(e)):
    if row == 0:
        di = True
    if row == n-1:
        di = False

    rail[row][col] = '0'
    col += 1

    if di:
        row += 1
    else:
        row -= 1

cnt = 0
for i in range(n):
    for j in range(len(e)):
        if rail[i][j] == '0' and cnt < len(e):
            rail[i][j] = e[cnt]
            cnt += 1

d = []
row, col = 0, 0
for i in range(len(e)):
    if row == 0:
        di = True
    if row == n-1:
        di = False

    if rail[row][col] != '0':
        d.append(rail[row][col])
        col += 1

    if di:
        row += 1
    else:
        row -= 1
d = ''.join(d)
print(d)